wormhole-connect/src/views/v2/Bridge/ReviewTransaction/GasSlider.tsx

```diff
@@ -0,0 +1,206 @@
+  const dispatch = useDispatch();
+  const theme = useTheme();
+
+  const { token: sourceToken, toChain: destChain } = useSelector(
+    (state: RootState) => state.transferInput,
+  );
+
+  const {
+    usdPrices: { data },
+  } = useSelector((state: RootState) => state.tokenPrices);
+
+  const prices = data || {};
+
+  const destChainConfig = config.chains[destChain!];
+  const sourceTokenConfig = config.tokens[sourceToken];
+  const nativeGasToken = config.tokens[destChainConfig!.gasToken];
+
+  const [isGasSliderOpen, setIsGasSliderOpen] = useState(!props.disabled);
+  const [percentage, setPercentage] = useState(0);
+
+  const [debouncedPercentage] = useDebounce(percentage, 500);
+
+  const ThumbWithTokenIcon = (props: React.HTMLAttributes<unknown>) => {
+    const { children, ...other } = props;
+    return (
+      <SliderThumb {...other}>
+        {children}
+        <TokenIcon icon={nativeGasToken.icon} height={16} />
+      </SliderThumb>
+    );
+  };
+
+  useEffect(() => {
+    dispatch(setToNativeToken(debouncedPercentage / 100));
+  }, [debouncedPercentage]);
+
+  const nativeGasPrice = useMemo(() => {
+    const tokenAmount = toFixedDecimals(
+      props.destinationGasFee?.toString() || '0',
+      6,
+    );
+    const tokenPrice = calculateUSDPrice(
+      toFixedDecimals(props.destinationGasFee?.toString() || '0', 6),
+      prices,
+      nativeGasToken,
+    );
+
+    return (
+      <Typography fontSize={14}>
+        {`${tokenAmount} ${getDisplayName(
+          sourceTokenConfig as TokenConfig,
+        )} ${tokenPrice}`}
+      </Typography>
+    );
+  }, [nativeGasToken, prices, props.destinationGasFee, sourceTokenConfig]);
+
+  // Checking required values
+  if (!sourceTokenConfig || !destChainConfig || !nativeGasToken) {
+    return <></>;
+  }
+
+  return (
+    <Card className={classes.card} variant="elevation">
+      <CardContent>
+        <Stack
+          direction="row"
+          alignItems="center"
+          justifyContent="space-between"
+        >
+          <Typography>Buy more gas for transactions</Typography>
+          <StyledSwitch
+            checked={isGasSliderOpen}
+            onClick={(e: any) => {
+              const { checked } = e.target;
+
+              setIsGasSliderOpen(checked);
+
+              if (!checked) {
+                setPercentage(0);
+                dispatch(setToNativeToken(0));
+              }
+            }}
+          />
+        </Stack>
+        <Collapse in={isGasSliderOpen} unmountOnExit>
+          <div className={classes.container}>
+            <Typography color={theme.palette.text.secondary} fontSize={14}>
+              Estimated amount needed for this transaction
+            </Typography>
+            <div>
+              <StyledSlider
+                slots={{ thumb: ThumbWithTokenIcon }}
+                aria-label="Native gas conversion amount"
+                defaultValue={0}
+                value={percentage}
+                baseColor="#C1BBF6"
```

Comment:
These color codes will eventually end up in theme overrides when doing the styling work.